prov/shm/src/smr.h

```diff
@@ -418,15 +418,13 @@ static inline int smr_cma_loop(pid_t pid, struct iovec *local,
 static inline struct smr_inject_buf *
 smr_get_txbuf(struct smr_region *smr)
 {
-	struct smr_inject_buf * txbuf = NULL;
+	struct smr_inject_buf * txbuf;
 
 	pthread_spin_lock(&smr->lock);
-	if (smr_freestack_isempty(smr_inject_pool(smr)))
-		goto out;
-
-	txbuf = smr_freestack_pop(smr_inject_pool(smr));
-
-out:
+	if (OFI_LIKELY(!smr_freestack_isempty(smr_inject_pool(smr))))
+		txbuf = smr_freestack_pop(smr_inject_pool(smr));
+	else
+		txbuf = NULL;
```

Comment:
I'm fine with an if/else here to assign tx_buf but using OFI_LIKELY/UNLIKELY is discouraged. Please drop

Reply:
Oh, I didn't realize LIKELY/UNLIKELY was discouraged.  Thanks.  I hope you don't mind, I merged this into your first commit :-)

Reply:
Perfect! Thanks!
Yeah in general these compiler optimizations can act a bit weird and can have the opposite effects if you don't REALLY know your code.
See [here](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-g_t_005f_005fbuiltin_005fexpect-4159) for more information, specifically the line:
You may use __builtin_expect to provide the compiler with branch prediction information. In general, you should prefer to use actual profile feedback for this (-fprofile-arcs), as programmers are notoriously bad at predicting how their programs actually perform.

Reply:
Exactly!  :)  Even in this case, we don't know if the inject pool will be empty or not when this is called, especially under load.  And if we get this wrong under load, we just made the situation worse.